src/sinks/azure_blob.rs

```diff
@@ -0,0 +1,832 @@
+        Encoding::Text => {
+            let mut bytes = log
+                .get(log_schema().message_key())
+                .map(|v| v.as_bytes().to_vec())
+                .expect("Failed to encode event as text");
+            bytes.push(b'\n');
+            bytes
+        }
+    };
+
+    Some(EncodedEvent::new(PartitionInnerBuffer::new(
+        bytes,
+        key.into(),
+    )))
+}
+
+fn build_request(
+    partition: PartitionInnerBuffer<Vec<u8>, Bytes>,
+    compression: Compression,
+    container_name: String,
+    blob_time_format: String,
+    blob_append_uuid: bool,
+) -> AzureBlobSinkRequest {
+    let (inner, key) = partition.into_parts();
+    let filename = {
+        let time_format = Utc::now().format(&blob_time_format);
+
+        if blob_append_uuid {
+            let uuid = Uuid::new_v4();
+            format!("{}-{}", time_format.to_string(), uuid.to_hyphenated())
+        } else {
+            time_format.to_string()
+        }
+    };
+    let blob = String::from_utf8_lossy(&key[..]).into_owned();
+    let blob = format!("{}{}.{}", blob, filename, compression.extension());
+
+    debug!(
+        message = "Sending events.",
+        bytes = ?inner.len(),
+        container = ?container_name,
+        blob = ?blob
+    );
+
+    AzureBlobSinkRequest {
+        container_name,
+        blob_data: inner,
+        blob_name: blob,
+        content_encoding: compression.content_encoding(),
+        content_type: compression.content_type(),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::collections::BTreeMap;
+
+    #[test]
+    fn generate_config() {
+        crate::test_util::test_generate_config::<AzureBlobSinkConfig>();
+    }
+
+    #[test]
+    fn azure_blob_encode_event_text() {
+        let message = String::from("hello world");
+        let event = Event::from(message.clone());
+        let blob_prefix = Template::try_from("logs/blob/%F").unwrap();
+        let encoding = EncodingConfig {
+            codec: Encoding::Text,
+            schema: None,
+            only_fields: None,
+            except_fields: None,
+            timestamp_format: None,
+        };
+
+        let bytes = encode_event(event, &blob_prefix, &encoding).unwrap();
+
+        let encoded_message = message + "\n";
+        let (bytes, _) = bytes.item.into_parts();
+        assert_eq!(&bytes[..], encoded_message.as_bytes());
+    }
+
+    #[test]
+    fn azure_blob_encode_event_json() {
+        let message = String::from("hello world");
+        let mut event = Event::from(message.clone());
+        event.as_mut_log().insert("key", "value");
+        let blob_prefix = Template::try_from("logs/blob/%F").unwrap();
+        let encoding = EncodingConfig {
+            codec: Encoding::Ndjson,
+            schema: None,
+            only_fields: None,
+            except_fields: None,
+            timestamp_format: None,
+        };
+
+        let bytes = encode_event(event, &blob_prefix, &encoding).unwrap();
```

Comment:
FTR creating `Event`s directly is a pattern we are slowly moving away from. Prefer something like the following:
```suggestion
        let mut log = LogEvent::from(message.clone());
        log.insert("key", "value");
        let blob_prefix = Template::try_from("logs/blob/%F").unwrap();
        let encoding = EncodingConfig {
            codec: Encoding::Ndjson,
            schema: None,
            only_fields: None,
            except_fields: None,
            timestamp_format: None,
        };

        let bytes = encode_event(log.into(), &blob_prefix, &encoding).unwrap();
```